### PR TITLE
Support string format for timestamp

### PR DESCRIPTION
**Pull request by lahsivjar**
Add support for string formatted timestamp.

## Motivation/summary

See [this comment 
](https://github.com/elastic/apm-server/pull/9349#issuecomment-1278968410)
## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
~~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~~
~~- [ ] Documentation has been updated~~

## How to test these changes

1. Create a request body for the `log` event type:
```json
{"metadata":{"process":{"pid":1234,"title":"/usr/lib/jvm/java-10-openjdk-amd64/bin/java","ppid":1,"argv":["-v"]},"system":{"architecture":"amd64","detected_hostname":"8ec7ceb99074","configured_hostname":"host1","platform":"Linux","container":{"id":"8ec7ceb990749e79b37f6dc6cd3628633618d6ce412553a552a0fa6b69419ad4"},"kubernetes":{"namespace":"default","pod":{"uid":"b17f231da0ad128dc6c6c0b2e82f6f303d3893e3","name":"instrumented-java-service"},"node":{"name":"node-name"}}},"service":{"name":"1234_service-12a3","version":"4.3.0","node":{"configured_name":"8ec7ceb990749e79b37f6dc6cd3628633618d6ce412553a552a0fa6b69419ad4"},"environment":"production","language":{"name":"Java","version":"10.0.2"},"agent":{"version":"1.10.0","name":"java","ephemeral_id":"e71be9ac-93b0-44b9-a997-5638f6ccfc36"},"framework":{"name":"spring","version":"5.0.0"},"runtime":{"name":"Java","version":"10.0.2"}},"labels":{"group":"experimental","ab_testing":true,"segment":5}}}
{"log": {"message": "test log message with timestamp", "@timestamp": "2006-01-02T15:04:05.999-0700"}}
```

2. Send a curl request with the above request body:
```bash
curl -i -H "Content-type: application/x-ndjson" -H "Authorization: Bearer {APM_SERVER_SECRET_TOKEN}" --data-binary @<request_body_path> {APM_SERVER_URL}:8200/intake/v2/events
```

## Related issues

Related to #9100

### CODE REVIEW

File: internal/model/modeldecoder/nullable/nullable.go [hunk 1]
```diff
@@ -77,12 +78,23 @@ func init() {
...
-		default:
+		case jsoniter.NumberValue:
 			us := iter.ReadInt64()
 			s := us / 1000000
 			ns := (us - (s * 1000000)) * 1000
 			(*((*TimeMicrosUnix)(ptr))).Val = time.Unix(s, ns).UTC()
 			(*((*TimeMicrosUnix)(ptr))).isSet = true
+		case jsoniter.StringValue:
+			tstr := iter.ReadString()
+			t, err := time.Parse("2006-01-02T15:04:05.999-0700", tstr)
```
Comment (axw) [inline internal/model/modeldecoder/nullable/nullable.go:89]: Can we use `time.RFC3339` here? In my brief test, it accepts the fractional (milliseconds) component.

Comment (lahsivjar) [reply internal/model/modeldecoder/nullable/nullable.go:89]: It fails to parse the timezone correctly, [example](https://go.dev/play/p/2W16kCESqyT)

Comment (axw) [reply internal/model/modeldecoder/nullable/nullable.go:89]: I see, thanks!

**Review by axw — APPROVED** [review]
LGTM!

**Comment by SylvainJuge** [comment]
Hi @lahsivjar , I tried checking this PR (combined with https://github.com/elastic/apm-server/pull/9349) on `main` and it seems that the following format is still not supported: `2022-10-20T08:36:43.789Z`, is that expected or not ?

I found this with a sample test application which might different from the one I used previously, hence the different timestamp format.

**Comment by lahsivjar** [comment]
@SylvainJuge It is expected, we are only allowing `strict_datetime`, please refer to [this comment chain](https://github.com/elastic/apm-server/pull/9349#issuecomment-1280314173) for details.

**Comment by lahsivjar** [comment]
Hmm, I have made a mistake here. It seems I added support for `2006-01-02T15:04:05.999-0700` instead of `yyyy-MM-dd'T'HH:mm:ss.SSSZ`. I will create a PR for the fix 🙇

**Comment by lahsivjar** [comment]
Created a [PR for the fix](https://github.com/elastic/apm-server/pull/9411).

**Pull request merged**